variables.tf

```diff
@@ -110,6 +110,12 @@ variable "logging" {
   type = object({
     target_bucket = string
     target_prefix = string
+    target_object_key_format = optional(object({
```

Comment:
I see you try to follow the terraform resource/api syntax in your variable as well and this has the upside:
* probably most future proof

but the downside:
* really hard to read.

So if we go to the most easily read option then it would probably be:

```
    target_object_key_format = optional(object({
      partitioned_prefix_data_source = optional(string, null)
      simple_prefix = optional(bool, false)
    }), {})
```

Which is then maybe not future proof enough. So how about we use something in between?:

```
variable "logging" {
  type = object({
    target_bucket = string
    target_prefix = string
    target_object_key_format = optional(object({
      partitioned_prefix = optional(object({
        partition_date_source = string
      }), null)
      simple_prefix = optional(bool, false)
    }), {})
  })
  default     = null
  description = "Logging configuration, logging is disabled by default."

  validation {
    condition = !(
      contains(keys(var.logging.target_object_key_format), "partitioned_prefix") &&
      contains(keys(var.logging.target_object_key_format), "simple_prefix")
    )
    error_message = "Invalid configuration: Both 'partitioned_prefix' and 'simple_prefix' cannot be used together."
  }

  validation {
    condition = (
      !contains(keys(var.logging.target_object_key_format), "partitioned_prefix") ||
      contains(["EventTime", "DeliveryTime"], var.logging.target_object_key_format.partitioned_prefix.partition_date_source)
    )
    error_message = "Invalid configuration: 'partition_date_source' must be either 'EventTime' or 'DeliveryTime' when 'partitioned_prefix' is specified."
  }
}
```

This solves for me the biggest issue with the variable, and that is the simple prefix, enabling means setting this to` {},` which is not intuitive for me and something thats not been done in any of our other modules :-) 

Added some validation as well

Reply:
Updated this to be more clear